src/app/graphql/resolvers.ts

```diff
@@ -0,0 +1,130 @@
+import prisma from '../client/client';
+import { UserInput } from '../interfaces';
+import bcrypt from 'bcrypt';
+import { CustomError } from '../errors/CustomError';
+
+const isPasswordValid = (password: string): boolean => {
+  const minLength = 6;
+  const hasLetter = /[a-zA-Z]/.test(password);
+  const hasDigit = /\d/.test(password);
+
+  return password.length >= minLength && hasLetter && hasDigit;
+};
+
+export const resolvers = {
+  Query: {
+    hello: () => 'hello!',
+    users: async () => {
+      const users = await prisma.user.findMany();
+
+      if (users.length == 0) {
+        throw new CustomError(
+          '404',
+          'Users not found!',
+          {
+            field: 'User',
+            reason: 'There are no users.',
+          }
+        );
+      }
+
+      return users;
+    },
+    user: async (_: unknown, { id }: { id: number }) => {
+        if(id == 0){
+        throw new CustomError(
+            '400',
+            'Invalid request!',
+            {
+              field: 'id',
+              reason: 'The id needs to be greater than 0!',
+            }
+        );
+      }
+
+      const user = prisma.user.findUnique({
+        where: { id: id },
+      });
+      if (user == null) {
+        throw new CustomError(
+            '404',
+            'User not found!',
+            {
+              field: 'id',
+              reason: 'The provided id does not exist.',
+            }
+        );
+      }
+
+      return user;
+    },
+  },
+  Mutation: {
+    createUser: async (_: unknown, { data }: UserInput) => {
+      if (!data) {
+        throw new CustomError(
+          '400',
+          'Data is missing!',
+          { 
+            field: 'data', 
+            reason: 'You need to input data!',
+          }
+        );
+      }
+
+      const { name, email, password, birthDate } = data;
+
+      if( !name || !email || !password || !birthDate ){
+        throw new CustomError(
+          '400',
+          'Invalid input!',
+          { 
+            field: 'data', 
+            reason: 'Name, email, password and birthDate are required!'
+          }
+        );
+      }
+
+      const existingEmail = await prisma.user.findUnique({
+        where: { email: email },
```

Comment:
depois tira um tempinho pra estudar sobre `Object Shorthand Notation`

```suggestion
        where: { email },
```

Reply:
sim sim!! obrigada pela sugestão!